README.md

```diff
@@ -43,17 +43,123 @@ requires = ['vulcan-py']
+By default, vulcan decides which python to use to generate a lockfile based on the currently active
+virtualenv. If there is not a currently active virtualenv, it will default to whichever version of python
+vulcan itself was installed with. Both of these behaviors can be overridden with the python-lock-with key,
+which specifies which python should be used to lock
+
+```toml
+[tool.vulcan]
+python-lock-with = "3.9"
+```
+
+### plugins
+
+Vulcan supports plugins, which can be called as a part of the build system to do some action on the
+in-progress build. These are registered via [entry points](https://github.com/optiver/vulcan-py#plugins), and
+to ensure there are not any accidental plugins activated they must be specified in the plugins config argument
+as well.
+
+```toml
+[tool.vulcan]
+plugins = ["plug1", "plug2"]
+```
+
+### shiv
+
+This repeatable section allows configuration of a shiv command to be called when invoking `vulcan build --shiv`.
+There are some mandatory keys, and some optional keys available.
+
+Note the double brackets, this is what makes the section repeatable.
+
+```toml
+[[tool.vulcan.shiv]]
+bin_name = "output_binary_name"    # mandatory
+console_script = "console_script_name"      #| Pick exactly one of
+entry_point = "path.to.entry_point:main"    #|
+interpreter = "/usr/bin/env python3.9"  # optional
+with_extras = ["extra1", "another_extra"]  # optional
+extra_args = "--any --other --shiv --arguments"
+```
+
+
+### dependencies
+
+This and the next section make the core of vulcan's functionality. These are the top-level unlocked
+dependencies for your application or library. With no-lock, these are translated directly into wheel
+dependencies without modification. Normally, these are used to determine the contents of the lockfile.
 
 ```toml
 [tool.vulcan.dependencies]
-toml='~=0.10'
-setuptools='~=53.0.0'
-wheel='~=0.36.2'
-dataclasses='~=0.8'
+requests=""  # no specific version
+lxml="~=4.0"  # with a version specification
+'build[virtualenv]' = ""   # with extras
+# build = {version="", extras=["virtualenv"]}  # different way of specifying extras
+```
+
+Note that if this section is specified, it is REQUIRED to put "dependencies" into the `dynamic` key in the
+`[project]` section of pyproject.toml
+
+### extras
+
+This section is similar to the above section, and is used to specify additional extras your application may
+provide. These are also used in resolving the lockfile if no-lock is not present, and it must be possible to
+install all extras at the same time.
+
+```toml
+[tool.vulcan.extras]
+extra1 = ["requests[socks]~=2.0"]
+extra2 = ["click"]
 ```
 
-Specify dependencies using the form: `package_name='{pep-508-spec}'`. Extras are supported in the package name.
+### dev-dependencies
+
+Finally, this section is used to specify any dependencies used for development, such as mypy or pytest. These
+are NOT used in resolving the lockfile, and are installed when `vulcan develop` is run. It is possible to only
```

Comment:
does this mean i still need to do `pip install -e .` beforehand or does `vulcan develop` by deafult also do that? If either, could you make that clear?

Reply:
Vulcan develop also does will add a bit.